lightning_client.go

```diff
@@ -2703,3 +2830,75 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// AcceptorFunction is the signature used for functions passed to our channel
+// acceptor.
+type AcceptorFunction func(*AcceptorRequest) (*AcceptorResponse, error)
+
+// AcceptChannel create a channel acceptor using the accept function
+// passed in.
+func (s *lightningClient) AcceptChannel(ctx context.Context,
+	accept AcceptorFunction) (chan error, error) {
+
+	acceptStream, err := s.client.ChannelAcceptor(
+		s.adminMac.WithMacaroonAuth(ctx),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	errChan := make(chan error, 1)
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		for {
+			request, err := acceptStream.Recv()
+			if err != nil {
+				errChan <- fmt.Errorf("channel acceptor "+
+					"receive failed: %v", err)
+
+				return
+			}
+
+			accReq, err := newAcceptorRequest(request)
+			if err != nil {
+				errChan <- fmt.Errorf("invalid request sent "+
+					"from lnd: %v", err)
+
+				return
+			}
+
+			resp, err := accept(accReq)
+			if err != nil {
+				errChan <- fmt.Errorf("accept funciton "+
+					"failed: %v", err)
+
+				return
+			}
+
+			rpcResp := &lnrpc.ChannelAcceptResponse{
+				Accept:          resp.Accept,
+				PendingChanId:   resp.PendingChanId,
```

Comment:
We could check if the `PendingChanId` in the response is empty and default to the value in the request. Or just remove it completely as @bhandras suggested. I think the field is only in the response to make sure the request and response are matched, since with streams it's possible to get requests and send responses out of order. And because we do it synchronously here, we could abstract that away from the `lndclient` user.

Reply:
see comment above!